src/components/sbb-navigation-section/sbb-navigation-section.scss

```diff
@@ -17,8 +17,8 @@
   --sbb-focus-outline-color: var(--sbb-focus-outline-color-dark);
   --sbb-navigation-section-width: 100%;
 
-  // We need to handle here the placement within the grid, since the host is exposed to the parent's grid (sbb-navigation)
-
+  // We need to handle here the placement and styling within the grid, since the host
```

Comment:
Did you test like display: unset or not including sbb.host-component-properties at all? Then maybe the grid styles also work if applied to an inner element.